onnx/inliner/inliner.cc

```diff
@@ -720,5 +722,61 @@ void InlineSelectedFunctions(
   }
   InlinerImpl::InlineSelectedFunctions(model, to_inline, schema_registry);
 }
+
+// Implementation of the Renamer class using InliningRenamer for proper graph-value attribute handling
+class Renamer::Impl : public InliningRenamer {
+ private:
+  std::unique_ptr<NameGenerator> generator_;
+
+ public:
+  Impl(const std::string& prefix, const GraphProto& graph)
+      : InliningRenamer("__" + prefix, *(generator_ = std::make_unique<NameGenerator>(graph))) {}
+
+  Impl(const std::string& prefix, const FunctionProto& function)
+      : InliningRenamer("__" + prefix, *(generator_ = std::make_unique<NameGenerator>(function))) {}
+
+  std::string BindToUniqueName(const std::string& original_name) {
+    // First create the unique name using the inherited MakeUnique method
+    std::string unique_name = MakeUnique(original_name);
+
+    // Then bind the original name to the unique name
+    if (!rename_scopes.empty()) {
```

Comment:
I think this condition is expected to be true? Maybe it is better to change the if-condition to an assert that rename_scopes is not empty?